src/test/kotlin/racing/application/RacingServiceTest.kt

```diff
@@ -1,27 +1,32 @@
 package racing.application
 
 import io.kotest.core.spec.style.BehaviorSpec
+import io.kotest.matchers.collections.shouldBeIn
 import io.kotest.matchers.collections.shouldHaveSize
 import io.kotest.matchers.shouldBe
 import racing.domain.Capacity
+import racing.domain.Cars
+import racing.domain.DirectionType
 import racing.domain.Distance
+import racing.domain.Name
 
 class RacingServiceTest : BehaviorSpec({
-    given("racing 메서드를 호출할 때 ") {
-        `when`("전진 전략이 항상 참을 반환하고") {
-            val racingService = RacingService { true }
+    Given("racing 메서드를 호출할 때 ") {
+        When("전진 전략이 항상 전진을 반환하고") {
+            val racingService = RacingService { DirectionType.STRAIGHT }
 
-            and("유효한 횟수와 자동차 대수를 인자로 전달할 경우 ") {
-                val info = RacingInfo(round = Capacity(5), capacity = Capacity(5))
+            And("유효한 횟수와 자동차 참가자를 인자로 전달할 경우 ") {
+                val info = RacingInfo(round = Capacity(5), cars = Cars("a,b,c,d,e"))
                 val actual = racingService.racing(info)
 
-                then("전달받은 자동차 대수만큼 자동차를 생성하고 결과를 반환한다.") {
-                    actual shouldHaveSize 5
-                    actual.mapIndexed { idx, info ->
+                Then("전달받은 자동차 대수만큼 자동차를 생성하고 결과를 반환한다.") {
+                    actual.racingHistories shouldHaveSize 5
+                    actual.racingHistories.mapIndexed { idx, info ->
                         info.records.forEach {
                             it.value shouldBe Distance(idx + 1L)
                         }
                     }
+                    actual.winners shouldBe arrayOf(Name("a"),Name("b"),Name("c"),Name("d"),Name("e"))
                 }
             }
```

Comment:
BDD 방식으로 테스트를 잘 작성해주신 것 같아요👍👍👍

저는 BDD 방식을 사용할 땐 [Cucumber에서 제공하는 예시](https://cucumber.io/docs/bdd/)들을 많이 참고하는 편이에요.

한솔님도 잘 아시고 계시겠지만 BDD에서 주로 사용하는 Given/When/Then은 일반적으로 다음과 같이 표현할 수 있는데요,
- Given: 주어진 조건, 상황
- When: 발생한 사건, 행위
- Then: 그로 인한 결과 

이를 바탕으로 위 코드를 다음과 같이 표현할 수도 있을 것 같아요. 

물론 테스트를 작성하는 스타일 차이지만..! 이런 방식도 있다는 정도로 생각해주시면 좋을 것 같습니다🙂
```suggestion
    Given("유효한 횟수와 자동차 참가자가 있다") {
        val info = RacingInfo(round = Capacity(5), cars = Cars("a,b,c,d,e"))
        And("전진 전략이 항상 전진이다") {
            val racingService = RacingService { DirectionType.STRAIGHT }
            When("경주를 진행하면") {
                val actual = racingService.racing(info)
                Then("경주 결과를 확인할 수 있다") {
                    // 검증
                }
            }
        }
    }
```

Reply:
given절에 계속 when 내용이 들어가는 것 같아 고민을 좀 하고 있었는데 리뷰어님이 제안해주신 부분이 좀 더 설득력이 있는 내용같습니다 ㅎㅎ